tests/test_createtag.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+import teachers_gitlab.main as tg
+
+def test_create_tag(mock_gitlab):
+    entries = [
+        {'login': 'alpha'},
+    ]
+    
+    mock_gitlab.register_project(452, 'student/alpha')
+    
+    mock_gitlab.on_api_post(
+        'projects/452/repository/tags',
+        request_json={
+            'ref': '', 
+            'tag_name': 'tag1'
+            },
+        response_json={
+        }
+    )
+
+    mock_gitlab.report_unknown()
+
+    tg.action_create_tag(
+        mock_gitlab.get_python_gitlab(),
+        logging.getLogger("createtag"),
+        tg.ActionEntries(entries),
+        'student/{login}',
+        'tag1',
+        '',
+        ''
+    )
+    
+def test_create_existing_tag(mock_gitlab):
+    entries = [
+        {'login': 'alpha'},
+    ]
+    
+    mock_gitlab.register_project(452, 'student/alpha')
+    
+    mock_gitlab.on_api_post(
+        'projects/452/repository/tags',
+        request_json={
+            'ref': 'double', 
+            'tag_name': 'tag2'
+            },
+        response_json={
+        }
+    )
+    
+    mock_gitlab.on_api_post(
+        'projects/452/repository/tags',
+        request_json={
+            'ref': 'double', 
+            'tag_name': 'tag2'
+            },
+        response_json={
+        }
+    )
+
+    mock_gitlab.report_unknown()
+
+    tg.action_create_tag(
+        mock_gitlab.get_python_gitlab(),
+        logging.getLogger("createtag"),
+        tg.ActionEntries(entries),
+        'student/{login}',
+        'tag2',
+        'double',
+        ''
+    )
+    
+    tg.action_create_tag(
+        mock_gitlab.get_python_gitlab(),
+        logging.getLogger("createtag"),
+        tg.ActionEntries(entries),
+        'student/{login}',
+        'tag2',
+        'double',
+        ''
+    )
```

Comment:
We are mocking everything - is it really needed to actually call `tg.action_create_tag` twice to emulate that the tag already exists? But maybe I am missing something very obvious here...